app/tasks/maintenance/remove_piece_justificative_file_not_visible_task.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+module Maintenance
+  class RemovePieceJustificativeFileNotVisibleTask < MaintenanceTasks::Task
+    attribute :procedure_id, :string
+    validates :procedure_id, presence: true
+
+    def collection
+      procedure = Procedure.with_discarded.find(procedure_id.strip.to_i)
+      procedure.dossiers.state_not_brouillon
+    end
+
+    def process(dossier)
+      dossier.champs.each do |champ|
+        next unless champ.piece_justificative_file.attached?
+        next if champ.visible?
+
+        champ.piece_justificative_file.purge_later
+      end
```

Comment:
un truc m'echappe p-e, pourquoi ne pas faire appel a la methode directement ? 
```suggestion
      dossier.each(&:remove_piece_justificative_file_not_visible!)
```

Reply:
oui tu as raison, c‘est qu‘au début j‘étais parti pour faire la collection sur les champs plutôt que sur les dossiers et puis j'ai changé d‘avis